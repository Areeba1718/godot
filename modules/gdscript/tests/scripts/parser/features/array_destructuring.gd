func test():
    print("Simple usage:")
    if true:
        var [var foo, var ..bar] = [123, "1234"]
        print(foo) # Should print "123"
        print(bar) # Should print "["1234"]"

    print("Complex usage:")
    if true:
        var [var a, var b, ..[var c, var d, ..[var e, var f]]] = [1, 2, 3, 4, 5, 6]
        prints(a, b, c, d, e, f) # Should print "1 2 3 4 5 6"

    print("Typed usage:")
    if true:
        var [var foo: int, var bar: String] = [123, "1234"]
        print(foo) # Should print "123"
        print(bar) # Should print "1234"

    print("Empty slot usage:")
    if true:
        var [_, var bar: String] = [123, "1234"]
        print(bar) # Should print "1234"

    print("Multiple lines usage:")
    if true:
        var [
            var foo: int,
            var bar: String
        ] = [123, "1234"]
        print(foo) # Should print "123"
        print(bar) # Should print "1234"

