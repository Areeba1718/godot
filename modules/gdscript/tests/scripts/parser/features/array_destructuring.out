GDTEST_OK
Simple usage:
123
["1234"]
Complex usage:
1 2 3 4 5 6
Typed usage:
123
1234
Empty slot usage:
1234
Multiple lines usage:
123
1234
